### PR TITLE
Fix Rubocop warnings.

### PR DESCRIPTION
**Pull request by rajachan**
- Rubocop warns against using EO* delimiters for heredocs.
  Using more meaningful ones.
- Use percentage notation for symbol arrays
- Force `zero_only` octal style (to allow `mode 0744`, etc.)
- Enumerate Gemfile packages in alphabetical order as expected by rubocop
- Adding Build status to README

Signed-off-by: Raghu Raja <craghun@amazon.com>

---
Down from 58 Rubocop offenses to one minor style issue, but there are no more warnings/errors that cause the build to fail.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,15 +3,15 @@ source 'https://rubygems.org'
 gem 'berkshelf'
 
 group :style do
+  gem 'foodcritic', '~> 6.0.1'
```
Comment (bwbarrett) [inline Gemfile]: just a nit, but why the reorder?

Comment (rajachan) [reply Gemfile]: Rubcop wants it in alphabetical ordering.

Comment (bwbarrett) [reply Gemfile]: ah; makes sense, although a comment in the commit would have saved me some typing :p.

Comment (rajachan) [reply Gemfile]: That's fair. WIll add more info to the commit message for history.

The EOF/H change was to make rubocop happy again. It filters out EO* delimiters and warns about using a more meaningful one.

**Pull request merged**